### PR TITLE
[IMP] hr: UX and onboarding improvements

### PR DESCRIPTION
**Pull request by Einar-eian**
A rework of the UX/UI for parts of the HR/Employee app which includes:
- Skills are now translatable, enabling the detection of skills on resumes written in languages other than English.
- Improved order on fields in employee form view.
- Adding/improving placeholders for multiple fields.
- Adding/improving/removing tooltips for multiple fields.
- Adding/improving No Content helpers for multiple views to make the intention of the view clearer.
- Adding and removing measurement fields for graph and pivot views to improve usability.
- Spelling improvements.
- Improved wording for multiple field strings.
- Hiding buttons on employee form when employee is archived.
- Creating activities on departments is now possible.
- On/Off-boarding: If the employee/manager/coach is not linked to a user we now go up the chain of management until we find a manager linked to a user. This should allow a more seamless user experience.
- Adding filter for employees in Contact app.
- A new tab "Payroll" has been added in the Employee form view to gather all the payroll information.
- A new field "Legal Name" has been added to accommodate employees whose legal names diverge from their preferred name.
- Replacing the "Name" field with the "Legal Name" field in various payroll and contract situations.

task-3930677

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by Bertrand2 — COMMENTED** [review]
Huge work! Overall a very good work :) Here are some comments/questions/nitpicking

File: addons/web_hierarchy/static/src/hierarchy_controller.xml [hunk 1]
```diff
@@ -27,10 +27,8 @@
                     <t t-component="searchBarToggler.component" t-props="searchBarToggler.props"/>
                 </t>
                 <t t-set-slot="default" t-slot-scope="layout">
-                    <t t-if="displayNoContent">
-                        <t t-call="web.ActionHelper">
-                            <t t-set="noContentHelp" t-value="props.info.noContentHelp"/>
-                        </t>
+                    <t t-if="model.resIds.length === 0 and props.info.noContentHelp" t-call="web.ActionHelper">
```
Comment (aab-odoo) [inline addons/web_hierarchy/static/src/hierarchy_controller.xml:30]: displayNoContent wasn't defined? There's another occurrence of the variable in the template (line 5)...

Comment (aab-odoo) [reply addons/web_hierarchy/static/src/hierarchy_controller.xml:30]: A test would be nice :-)

File: addons/hr/models/hr_department.py [hunk 2]
```diff
@@ -8,7 +8,7 @@
 class Department(models.Model):
     _name = "hr.department"
     _description = "Department"
-    _inherit = ['mail.thread']
+    _inherit = ['mail.thread', 'mail.activity.mixin']
```
[the remark this answers is not in the document]
Comment (Einar-eian) [reply addons/hr/models/hr_department.py:11]: This is to enable activities to be added to departments

File: addons/hr/models/hr_employee_base.py [hunk 3]
```diff
@@ -42,7 +42,7 @@ class HrEmployeeBase(models.AbstractModel):
         ("home", "Home"),
         ("office", "Office"),
         ("other", "Other")], compute="_compute_work_location_name_type")
-    user_id = fields.Many2one('res.users')
+    user_id = fields.Many2one('res.users', help="")
```
[the remark this answers is not in the document]
Comment (Einar-eian) [reply addons/hr/models/hr_employee_base.py:45]: The idea was more to remove the tooltip text but if you know of a better way to do it I'm all ears!

**Comment by tivisse** [comment]
@robodoo r+

**Comment by aab-odoo** [comment]
This PR introduces a crash in the Planning app (in debug). It's a owl props validation error. Spotted by the nightly clickall: <https://runbot.odoo.com/runbot/build/68356574> @Einar-eian @tivisse

**Pull request closed without merging**